jcl/src/openj9.sharedclasses/share/classes/com/ibm/oti/shared/SharedClassURLHelperImpl.java

```diff
@@ -38,6 +40,9 @@
 	/* Not public - should only be created by factory */
 	private boolean minimizeUpdateChecks;
 	
+	/* Used to keep track of new jar files */
+	private Set<String> jarFileNameCache = ConcurrentHashMap.newKeySet();
```

Comment:
This needs to be lazily initialized to avoid growing footprint. If minimizeUpdateChecks is set, it should never be initialized.

Also since findSharedClass() and storeSharedClass() are synchronized, using a HashSet is fine.

Reply:
Already moved the initialization with HashSet to `newJarFileCheck()`. So the initialization occurs only when `newJarFileCheck()` is called for the first time (minimizeUpdateChecks is true in such case).